metric/metric.go

```diff
@@ -2,26 +2,161 @@ package metric
 
 import (
 	"strconv"
+	"strings"
+	"time"
 
 	"github.com/TwiN/gatus/v3/core"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 var (
+	namespace string
 	// This will be initialized once PublishMetricsForEndpoint.
 	// The reason why we're doing this is that if metrics are disabled, we don't want to initialize it unnecessarily.
-	resultCount *prometheus.CounterVec = nil
+	resultCount         *prometheus.CounterVec
+	resultSuccessGauge  *prometheus.GaugeVec
+	resultDurationGauge *prometheus.GaugeVec
+
+	resultDNSReturnCode *prometheus.CounterVec
+
+	resultTCPConnected *prometheus.CounterVec
+
+	resultStartTLSConnected                          *prometheus.CounterVec
+	resultStartTLSSSLLastChainExpiryTimestampSeconds *prometheus.GaugeVec
+
+	resultTLSConnected                          *prometheus.CounterVec
+	resultTLSSSLLastChainExpiryTimestampSeconds *prometheus.GaugeVec
+
+	resultHTTPConnected                          *prometheus.CounterVec
+	resultHTTPStatusCode                         *prometheus.CounterVec
+	resultHTTPSSLLastChainExpiryTimestampSeconds *prometheus.GaugeVec
 )
 
+func ensurePrometheusMetrics() {
+	if namespace == "" {
+		namespace = "gatus"
```

Comment:
This is unnecessarily confusing. I'd rather have `namespace` as a const and have a bool like `initializedMetrics`

Reply:
added initializedMetrics var for this usage